### PR TITLE
PyPIM integration

### PR DESCRIPTION
**Pull request by plule-ansys**
This PR integrates [PyPIM](https://pypim.docs.pyansys.com/). In an environment configured for PIM, DpfServer is now able to start a remote instance of DPF and connect to it.

The integration approach is intended to follow the current pydpf direction and does not add any gRPC specific dependency. PyPIM itself being based on gRPC, it is lazy loaded under the condition that the user is starting dpf in a pypim environment. It however changes the construction of the channel from "ip/port" to any address as supported by gRPC: https://grpc.github.io/grpc/core/md_doc_naming.html

It currently comes with a limitation that the headers are not transmitted to the instances. It means that this implementation will not work if the DPF instance is behind a gateway, or in an authenticated environment.

### CODE REVIEW

File: ansys/dpf/core/server.py [hunk 1]
```diff
@@ -351,25 +354,31 @@ def __init__(
...
             raise OSError("DPF does not support Ubuntu")
         elif launch_server:
-            self._server_id = launch_dpf(str(ansys_path), ip, port,
-                                         docker_name=docker_name,
-                                         timeout=timeout)
+            if is_pypim_configured() and not ansys_path and not docker_name:
+                self._remote_instance = launch_remote_dpf()
+                address = self._remote_instance.services["grpc"].uri
+            else:
+                ip = ip or LOCALHOST
```
Comment (anslpa) [inline ansys/dpf/core/server.py:366]: @plule-ansys I think this must be removed

File: ansys/dpf/core/server.py [hunk 2]
```diff
@@ -351,25 +354,31 @@ def __init__(
...
+        self.channel = grpc.insecure_channel(address)
 
         # assign to global channel when requested
         if as_global:
             dpf.core.SERVER = self
 
         # TODO: add to PIDs ...
 
-        # store port and ip for later reference
-        self._input_ip = ip
```
Comment (anslpa) [inline ansys/dpf/core/server.py:371]: @plule-ansys would it eventually be possible to keep self._input_ip and self._input_port at None ?

File: ansys/dpf/core/server.py [hunk 3]
```diff
@@ -351,25 +354,31 @@ def __init__(
...
+                address = self._remote_instance.services["grpc"].uri
+            else:
+                ip = ip or LOCALHOST
+                port = port or DPF_DEFAULT_PORT
+                self._server_id = launch_dpf(str(ansys_path), ip, port,
+                                            docker_name=docker_name,
+                                            timeout=timeout)
 
-        self.channel = grpc.insecure_channel("%s:%d" % (ip, port))
+        self.channel = grpc.insecure_channel(address)
```
Comment (anslpa) [inline ansys/dpf/core/server.py:372]: @plule-ansys see in later packages if the client constructor with full address can be used

File: ansys/dpf/core/server.py (hunk 1 above)
Comment (plule-ansys) [reply ansys/dpf/core/server.py]: oops, done thanks

File: ansys/dpf/core/server.py (hunk 2 above)
Comment (plule-ansys) [reply ansys/dpf/core/server.py:371]: Sure done. I'm keeping them as-is in the general case, and setting them to "None" when an address is provided.

File: ansys/dpf/core/server.py (hunk 3 above)
Comment (plule-ansys) [reply ansys/dpf/core/server.py:373]: Yes, hopefully the full address constructor can be used instead of the ip/port...

**Review by cbellot000 — APPROVED** [review]

**Review by anslpa — APPROVED** [review]

**Comment by PProfizi** [comment]
@plule-ansys Maybe a small test to pass the brand new codecov criteria?

**Pull request merged**